src/lib/db.ts

```diff
@@ -519,6 +520,15 @@ export class DB {
     });
   }
 
+  public copyOtherDocument(document: DocumentModelType, newTitle?: string) {
+    const title = `${newTitle || document.title || this.stores.problem.title || "Untitled"}`;
+    const content = cloneContentWithUniqueIds(document.content);
+    if (document.type === ProblemDocument) {
+      return this.createPersonalDocument({ title, content });
+    }
+    return this.createOtherDocument(document.type as OtherDocumentType, { title, content });
```

Comment:
```suggestion
    const copyType = document.type === ProblemDocument ? PersonalDocument : document.type as OtherDocumentType;
    return this.createOtherDocument(copyType, { title, content });
```
this way you don't need the `if` clause above.